### ISSUE TITLE
The microphone icon in Chrome stays open when disconnected

### ISSUE THREAD

**Issue by hookedupjoe** [hvianna/audioMotion-analyzer#52]
After stopping the use of the microphone, in Chrome, the mic icon stays on.  This happens in my project and the examples on the websites as well.   A solution is to also close the microphone track that is open when disconnecting.  

An example solution is to add an option to the disconnect method to optionally close the track. This made the icon go away when used and should be backward compatible for anyone still needing it to not be closed when disconnecting.

```
/**
	 * Disconnects audio sources from the analyzer
	 *
	 * @param [{object|array}] a connected AudioNode object or an array of such objects; if undefined, all connected nodes are disconnected
	 * @param {boolean} [stopTracks] if true, stops the track when disconnected
	 */
	disconnectInput( sources, stopTracks ) {
		if ( ! sources )
			sources = Array.from( this._sources );
		else if ( ! Array.isArray( sources ) )
			sources = [ sources ];
		for ( const node of sources ) {
			const idx = this._sources.indexOf( node );			
			if (stopTracks && node.mediaStream){
				for ( const ats of node.mediaStream.getAudioTracks() ){
					ats.stop();
				}
			}
			if ( idx >= 0 ) {
				node.disconnect( this._input );
				this._sources.splice( idx, 1 );
			}
		}
	}
```

Naturally can still be called with default sources:
`audioMotion.disconnectInput(false, true);`

**Comment by hvianna** [comment]
Thanks for this! Although the user could do that on their side, I think it's an elegant solution.

**Comment by hookedupjoe** [comment]
> Thanks for this! Although the user could do that on their side, I think it's an elegant solution.

Agreed - I just had to hunt down how to do it and that took a bit of doing so maybe this would help others.  Thanks for the killer awesome fantastic .. not enough good words to describe ... library.

**Comment by Staijn1** [comment]
I've noticed the same behavior of the mic staying open in both Firefox & Chrome in my personal project where I use this library. I figured I did something wrong and since I'm the only person using it.. I could not be bothered to fix it.
I'd love to see the code-snippet merged.

Although, should the track also be stopped when the analyzer is toggled off, using `toggleAnalyzer()`? (Preventing the icon being shown when the analyzer is toggled off)
Or is it then impossible to resume the same track after toggling the analyzer on again? I'm not familiar enough with using media devices to tell.

If desired I can raise a PR that toggles the track on `disconnectInput()` & `toggleAnalyzer()`

**Comment by hvianna** [comment]
I'll be adding this to v4.2.0

@Staijn1: according to [MDN docs](https://developer.mozilla.org/en-US/docs/Web/API/MediaStreamTrack/stop) the track is permanently ended on `stop()` and disassociated from the source stream, so it can't be resumed.

**Comment by hvianna** [comment]
Now available in [v4.2.0](https://github.com/hvianna/audioMotion-analyzer/releases/tag/4.2.0) 🎉
